### PR TITLE
Change `TIMESTAMP` and `TIME` parsing so that time zone information is preserved

### PR DESCRIPTION
**Pull request by AugustoFKL**
Fixing `TIME` and `TIMESTAMP` printing considering the input ([1](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#datetime-type)), instead of assuming that:
- The absence of arguments is equal to `WITHOUT TIME ZONE` (the standard recommends this, but may not be followed by all dialects)
- The presence of `WITH TIME ZONE` equals adding `TZ` to the name. I.e.: `TIMESTAMP WITH TIME ZONE` to `TIMESTMAPTZ`. For example, Oracle doesn't apply that logic ([2](https://docs.oracle.com/en/database/oracle/oracle-database/12.2/nlspg/datetime-data-types-and-time-zone-support.html#GUID-3F1C388E-C651-43D5-ADBC-1A49E5C2CA05)), but PostgreSQL does ([3](https://www.postgresql.org/docs/current/datatype-datetime.html)).

[1] : https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#datetime-type
[2] : https://docs.oracle.com/en/database/oracle/oracle-database/12.2/nlspg/datetime-data-types-and-time-zone-support.html#GUID-3F1C388E-C651-43D5-ADBC-1A49E5C2CA05
[3] : https://www.postgresql.org/docs/current/datatype-datetime.html
Resolves: https://github.com/sqlparser-rs/sqlparser-rs/issues/640
Fixes: https://github.com/sqlparser-rs/sqlparser-rs/pull/589

### CODE REVIEW

**Comment by alamb** [comment]
cc @waitingkuo

File: src/parser.rs [hunk 1]
```diff
@@ -5256,23 +5261,78 @@ mod tests {
...
+            "VARBINARY(20)",
+        );
+
+        fn test_parse_data_type(input: &str, expected_type: DataType, expected_str: &str) {
             all_dialects().run_parser_method(input, |parser| {
-                let data_type = parser.parse_data_type().unwrap().to_string();
-                assert_eq!(data_type, expected);
+                let data_type = parser.parse_data_type().unwrap();
+                assert_eq!(data_type, expected_type);
+                assert_eq!(expected_type.to_string(), expected_str.to_string());
```
Comment (alamb) [inline src/parser.rs:5335]: 👍

**Review by alamb — APPROVED** [review]
This looks great -- thank you @AugustoFKL

**Comment by waitingkuo** [comment]
@alamb @AugustoFKL I just reviewed the codes, it looks great! 👍  
sorry for the late reply

i'll file a issue to datafusion to align with this

**Pull request merged**